androidgen/src/main/java/com/azure/autorest/android/template/AndroidServiceClientBuilderTemplate.java

```diff
@@ -197,4 +232,133 @@ protected String primaryBuildMethodName(JavaSettings settings) {
+                                .description("The Azure Core generic ServiceClient Builder.")
+                                .isFinal(false)
+                                .wireType(ClassType.AndroidRestClientBuilder)
+                                .name("serviceClientBuilder")
+                                .isRequired(true)
+                                .isConstant(false)
+                                .fromClient(true)
+                                .annotations(new ArrayList<>())
+                                .defaultValue("new ServiceClient.Builder()")
+                                .build();
+                    } else {
+                        return p;
+                    }
+                }).collect(Collectors.toList());
+    }
+
+    private ClientMethodParameter credentialInterceptorParameter() {
+        return new ClientMethodParameter.Builder()
+                .description("The Interceptor to set intercept request and set credentials.")
+                .isFinal(false)
+                .wireType(ClassType.AndroidOkHttpInterceptor)
+                .name("credentialInterceptor")
+                .isRequired(true)
+                .isConstant(false)
+                .fromClient(true)
+                .annotations(new ArrayList<>())
+                .build();
+    }
+
+    private static void writeBuilderProperty(JavaSettings settings,
+                                      JavaClass classBlock,
+                                      String builderName,
+                                      String propDescription,
+                                      String propName,
+                                      IType propType) {
+        classBlock.blockComment(settings.getMaximumJavadocCommentWidth(), comment ->
+        {
+            comment.line(propDescription);
+        });
+        classBlock.privateMemberVariable(String.format("%1$s %2$s", propType, propName));
+
+        classBlock.javadocComment(comment ->
+        {
+            comment.description(String.format("Sets %1$s", propDescription));
+            comment.param(propName, String.format("the %1$s value.", propName));
+            comment.methodReturns(String.format("the %1$s", builderName));
+        });
+        classBlock.publicMethod(String.format("%1$s %2$s(%3$s %4$s)",
+                builderName,
+                CodeNamer.toCamelCase(propName),
+                propType,
+                propName), function ->
+        {
+            function.line(String.format("this.%1$s = %2$s;", propName, propName));
+            function.methodReturn("this");
+        });
+    }
+
+    private void fixImportForBuilderParameters(List<ClientMethodParameter> parameters, Set<String> imports) {
+        parameters.stream()
+                .forEach(p -> {
+                    fixImportIfBuilderParameter(p, imports);
+                });
+    }
+
+    private void fixImportIfBuilderParameter(ClientMethodParameter parameter, Set<String> imports) {
```

Comment:
I agree, unfortunately, the core gen doesn't have support to indicate that an IType is child type, so this import fixing. Didn't spend much time designing that in the core since there is only one child type scenario today, lets circle back once the high priority items to unblock chat is done.